### PR TITLE
PP-8736 - Create deploy GH action

### PR DESCRIPTION
**Pull request by lucas-av7**
# Changes

Create the deploy GH action for main, staging and production

# Notes

Ticket: https://jira.slc.efscloud.net/browse/PP-8736

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          role-to-assume: ${{ vars.IAM_ROLE_GITHUB }}
+          role-session-name: GitHub-Action-Role
+          aws-region: ${{ vars.AWS_REGION }}
+      - name: Deploying files
+        env:
+          S3_BUCKET: ${{ env.S3_BUCKET }}
+        run: |
+          aws s3 cp ./dist/ s3://$S3_BUCKET/ --recursive --only-show-errors
+          aws s3 cp ./specs/ s3://$S3_BUCKET/ --recursive --only-show-errors
+          aws s3 cp ./index.html s3://$S3_BUCKET/ --only-show-errors
```
Comment (axbgrossi) [inline .github/workflows/deploy.yml:60]:
```suggestion
          echo "Deploying index.html to s3://$S3_BUCKET/"
          aws s3 cp ./index.html s3://$S3_BUCKET/ --only-show-errors
          echo "S3 deployment complete."
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Invalidate CloudFront Cache
+        env:
+          SOURCE_PATH: '/*'
+          AWS_REGION:  ${{ env.AWS_REGION }}
+          DISTRIBUTION_ID: ${{ env.DISTRIBUTION_ID }}
+        run: |
+          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH --region $AWS_REGION
```
Comment (axbgrossi) [inline .github/workflows/deploy.yml:67]:
```suggestion
          echo "Invalidating CloudFront cache..."
          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH --region $AWS_REGION
          echo "CloudFront cache invalidation complete."
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          if [ "${{ github.ref_name }}" = "production" ]; then
+            echo "env_name=Production" >> $GITHUB_OUTPUT
+            echo "using Production environment"
+          elif [ "${{ github.ref_name }}" = "staging" ]; then
+            echo "env_name=Staging" >> $GITHUB_OUTPUT
+            echo "using Staging environment"
+          else
+            echo "env_name=Development" >> $GITHUB_OUTPUT
+            echo "using Development environment"
+          fi
```
Comment (axbgrossi) [inline .github/workflows/deploy.yml:36]:
```suggestion
          fi
          echo "Environment setup complete."
```

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+          echo "S3 deployment complete."
+      - name: Invalidate CloudFront Cache
+        env:
+          SOURCE_PATH: '/*'
+          AWS_REGION:  ${{ env.AWS_REGION }}
+          DISTRIBUTION_ID: ${{ env.DISTRIBUTION_ID }}
+        run: |
+          echo "Invalidating CloudFront cache..."
+          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH --region $AWS_REGION
+          echo "CloudFront cache invalidation complete."
```
Comment (axbgrossi) [inline .github/workflows/deploy.yml:82]:
```suggestion
          echo "Invalidating CloudFront cache..."
          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH --region $AWS_REGION &&\
          echo "CloudFront cache invalidation complete." ||\
          echo "CloudFront cache invalidation failed."
```

**Review by axbgrossi — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          aws s3 cp ./index.html s3://$S3_BUCKET/ --only-show-errors
+          echo "S3 deployment complete."
+      - name: Invalidate CloudFront Cache
+        env:
+          SOURCE_PATH: '/*'
+          AWS_REGION:  ${{ env.AWS_REGION }}
+          DISTRIBUTION_ID: ${{ env.DISTRIBUTION_ID }}
+        run: |
+          echo "Invalidating CloudFront cache..."
+          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH --region $AWS_REGION
```
Comment (sam-axcient) [inline .github/workflows/deploy.yml:81]:
```suggestion
          aws cloudfront create-invalidation --distribution-id $DISTRIBUTION_ID --paths $SOURCE_PATH
```
specifying a region is redundant as they are already set by `Configuring AWS credentials` step

**Review by sam-axcient — COMMENTED** [review]
LGTM

**Pull request merged**